tests/blockdev_test.go

```diff
@@ -0,0 +1,334 @@
+			 "model": null, "serial": null, "size": "1G", "state": null, "owner": "root", "group": "disk",
+			 "mode": "brw-rw----", "alignment": "0", "min-io": "512", "opt-io": "0", "phy-sec": "512", "log-sec": 256,
+			 "rota": "1", "sched": "mq-deadline", "rq-size": "256", "type": "part", "disc-aln": "0", "disc-gran": "0B",
+			 "disc-max": "0B", "disc-zero": "0", "wsame": "0B", "wwn": "0x61866da068d6a500249e4f2909cce7dc",
+			 "rand": "1", "pkname": "/dev/sdg", "hctl": null, "tran": null, "subsystems": "block:scsi:pci",
+			 "rev": null, "vendor": null, "zoned": "none"},
+			 {"name": "/dev/sdg2", "kname": "/dev/sdg2", "maj:min": "8:98", "fstype": "swap", "mountpoint": "[SWAP]",
+			 "label": null, "uuid": "3b1af9c7-740c-4f84-bd45-6d6f0fe848a5", "parttype": "0x82", "partlabel": null,
+			 "partuuid": "21d9a56e-02", "partflags": null, "ra": "128", "ro": 1, "rm": "0", "hotplug": "0",
+			 "model": null, "serial": null, "size": "4G", "state": null, "owner": "root", "group": "disk",
+			 "mode": "brw-rw----", "alignment": "0", "min-io": "512", "opt-io": "0", "phy-sec": "512", "log-sec": 256,
+			 "rota": "1", "sched": "mq-deadline", "rq-size": "256", "type": "part", "disc-aln": "0", "disc-gran": "0B",
+			 "disc-max": "0B", "disc-zero": "0", "wsame": "0B", "wwn": "0x61866da068d6a500249e4f2909cce7dc",
+			 "rand": "1", "pkname": "/dev/sdg", "hctl": null, "tran": null, "subsystems": "block:scsi:pci", "rev": null,
+			  "vendor": null, "zoned": "none"},
+			 {"name": "/dev/sdg3", "kname": "/dev/sdg3", "maj:min": "8:99", "fstype": "xfs", "mountpoint": "/",
+			 "label": null, "uuid": "333bb3f4-1151-4aaf-a988-72b49e4e2ec2", "parttype": "0x83", "partlabel": null,
+			 "partuuid": "21d9a56e-03", "partflags": null, "ra": "128", "ro": 1, "rm": "0", "hotplug": "0",
+			 "model": null, "serial": null, "size": "186.3G", "state": null, "owner": "root", "group": "disk",
+			 "mode": "brw-rw----", "alignment": "0", "min-io": "512", "opt-io": "0", "phy-sec": "512", "log-sec": 256,
+			 "rota": "1", "sched": "mq-deadline", "rq-size": "256", "type": "part", "disc-aln": "0", "disc-gran": "0B",
+			 "disc-max": "0B", "disc-zero": "0", "wsame": "0B", "wwn": "0x61866da068d6a500249e4f2909cce7dc",
+			 "rand": "1", "pkname": "/dev/sdg", "hctl": null, "tran": null, "subsystems": "block:scsi:pci",
+			 "rev": null, "vendor": null, "zoned": "none"}
+		  ]
+	   }
+	]
+}
+EOF
+`)
+
+func setupBlockDev(c *TestContext) error {
+	if err := ioutil.WriteFile(testFileName, jsondata, 0755); err != nil {
+		return err
+	}
+
+	if err := Setenv("SKYDIVE_AGENT_TOPOLOGY_BLOCKDEV_LSBLK_PATH", testFileName); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func TestBlockDevSimple(t *testing.T) {
+	test := &Test{
+		mode: Replay,
+
+		checks: []CheckFunction{func(c *CheckContext) error {
+			gremlin := c.gremlin.V().Has("Type", "blockdev", "Manager", "blockdev")
+
+			nodes, err := c.gh.GetNodes(gremlin)
+			if err != nil {
+				return err
+			}
+
+			if len(nodes) > 0 {
```

Comment:
len(nodes) == 0 ?

Reply:
I was missing:

setupFunction: setupBlockDev

I updated the test.